include/mbedtls/ccm.h

```diff
@@ -61,6 +61,8 @@
 #define MBEDTLS_ERR_CCM_BAD_INPUT       -0x000D
 /** Authenticated decryption failed. */
 #define MBEDTLS_ERR_CCM_AUTH_FAILED     -0x000F
+/** CCM functions called in the wrong sequence. */
```

Comment:
In one of previous reviews you said "I think we need a new error code here for calling CCM functions out of sequence or with lengths that don't match what was passed to set_lengths()."
Are you suggesting that I should remove the BAD_SEQUENCE error and use only BAD_INPUT?

Reply:
I think using BAD_INPUT for something like `update(); update_ad();` is a bit of a stretch, which is why I suggested BAD_SEQUENCE. On the other hand, the API functions don't always have enough information to distinguish between e.g. calling a function at the wrong time and passing too much input, which means that sometimes the choice of BAD_SEQUENCE vs BAD_INPUT can be misleading. So I'm ok with a single error code, and I'm ok with two error codes, but if there are two error codes, the code and the documentation need to be made clearer so that the distinction between these two errors is easy to understand. I think it's easier to stick to a single error code.